sphinxgallery/gen_rst.py

```diff
@@ -118,133 +78,161 @@ def flush(self):
+            if 20*'#' in line:
+                continue_text = True
+                if len(block) > 1:
 
-            fhindex += _thumbnail_div(target_dir, fname, snippet)
-            fhindex += """
 
-.. toctree::
-   :hidden:
+                    blocks.append(('code', block))
+                    block = '"""'
+                continue
+        if line.startswith('#') and continue_text:
+            block += line[2:]
+            continue
+        if not line.startswith('#') and continue_text:
 
-   /%s/%s\n""" % (target_dir, fname[:-3])
+            block += '"""'
+            blocks.append(('text', block))
+            block = ''
+            continue_text = False
 
+        # code blocks
+        block += line
 
-# clear at the end of the section
-    fhindex += """.. raw:: html\n
-    <div style='clear:both'></div>\n\n"""
+    # close  last block
+    if len(block) > 1:
+        if continue_text:
+            blocks.append(('text', block))
+        else:
+            blocks.append(('code', block))
+
+    f.close()
+    return blocks
+
+
+def codestr2rst(codestr):
+    """Return reStructuredText code block from code string"""
+    code_directive = "\n.. code-block:: python\n\n"
+    indented_block = '    ' + codestr.replace('\n', '\n    ')
+    return code_directive + indented_block
 
-    return fhindex
+
+def extract_intro(filename):
+    """ Extract the first paragraph of module-level docstring. max:95 char"""
+
+    first_text = split_code_and_text_blocks(filename)[0][-1]
+
+    paragraphs = first_text.split('\n\n')
+    if len(first_text) > 1:
+        first_par = re.sub('\n', ' ', paragraphs[1])
+        first_par = ((first_par[:95] + '...')
+                     if len(first_par) > 95 else first_par)
+    else:
+        raise ValueError("Missing first paragraph."
+                         "Please check the layout of your"
+                         " example file:\n {}\n and make sure"
+                         " it's correct.".format(filename))
+
+    return first_par
+
+
+def _plots_are_current(src_file, image_file):
+    """Test existence of image file and later touch time to source script"""
+
+    first_image_file = image_file.format(1)
+    needs_replot = (not os.path.exists(first_image_file) or
+              os.stat(first_image_file).st_mtime <= os.stat(src_file).st_mtime)
+    return not needs_replot
+
+
+def save_figures(image_path, fig_count):
+    """Save all open matlplotlib figures of the example code-block
+
+    Parameters
+    ----------
+    image_path : str
+        Path where plots are saved (format string which accepts figure number)
+    fig_count : int
+        Previous figure number count. Figure number add from this number
+
+    Returns
+    -------
+    list of strings containing the full path to each figure
+    """
+    figure_list = []
+    # In order to save every figure we have two solutions :
+    # * iterate from 1 to infinity and call plt.fignum_exists(n)
```

Comment:
Is this part of the comment still useful? It doesn't seem like this plt.fignum_exists is used below so I would remove it.